sparql/parse.go

```diff
@@ -102,6 +110,9 @@ func (query *Query) Parse(src io.Reader) error {
 					verb = ""
 				}
 			} else {
+				if namedGraph && query.NamedGraph == "" {
+					query.NamedGraph = string(b[uriStart+1 : s.Position.Offset])
+				}
```

Comment:
This function is getting to be fairly long. What do you think about breaking it up? If time is of the essence, I'd also be happy to create a refactoring ticket.

Reply:
Yeah, this is getting a bit long and I think needs cleanup. I mostly wanted to capture the current state here but can continue to clean it up.